python/taichi/misc/gui.py

```diff
@@ -123,30 +123,53 @@ def line(self, begin, end, radius, color):
         self.canvas.path_single(begin[0], begin[1], end[0], end[1], color,
                                 radius)
 
-    def show(self, file=None):
+    def show(self, file_or_image=None):
         self.core.update()
-        if file:
-            self.core.screenshot(file)
+        if isinstance(file_or_image, str):
+            self.core.screenshot(file_or_image)
+        elif file_or_image is not None:
+            self.set_image(file_or_image)
         self.clear(self.background_color)
 
-    def has_key_event(self):
-        return self.core.has_key_event()
 
     def get_event(self, *filter):
+        types = []
+        keys = []
+        combs = []
+        for ent in filter:
+            if ent in [GUI.PRESS, GUI.RELEASE]:
+                types.append(ent)
+            elif isinstance(ent, tuple) or isinstance(ent, list):
+                type, key = ent
+                combs.append((type, key))
```

Comment:
Just `combs.append(tuple(ent))`?

Reply:
We want to make sure that ent is a `(type, key)` instead of `(233, 666, 888)`.